arch/arm/core/isr_wrapper.S

```diff
@@ -128,12 +160,17 @@ _idle_state_cleared:
 	mov lr, r3
 #elif defined(CONFIG_ARMV7_M_ARMV8_M_MAINLINE)
 	pop {r0, lr}
+#elif defined(CONFIG_ARMV7_R)
 #else
 #error Unknown ARM architecture
 #endif /* CONFIG_ARMV6_M_ARMV8_M_BASELINE */
 
+#if defined(CONFIG_CPU_CORTEX_R)
+	mov r0, #RET_FROM_IRQ
+#endif
+
 	/* Use 'bx' instead of 'b' because 'bx' can jump further, and use
 	 * 'bx' instead of 'blx' because exception return is done in
 	 * _IntExit() */
-	ldr r0, =_IntExit
-	bx r0
+	ldr r1, =_IntExit
```

Comment:
Any reason this was changed @bbolen ?

Reply:
Exiting from an interrupt and exiting from a svc call require slightly different paths in arch/arm/core/exc_exit.S:z_ExcExit().  I'm using r0 to pass which context it is leaving from in z_ExcExit().  I could have left that code alone and just passed the return context in r1, but that seemed a little weird having an argument coming into z_ExcExit in r1, but nothing in r0.